numpy/__init__.pyi

```diff
@@ -83,7 +109,8 @@ class dtype:
     def type(self) -> builtins.type: ...
 
 
-_dtype_class = dtype      # for ndarray type
+_DtypeLike = Union[dtype, _ConvertibleToDtype]
+_Dtype = dtype  # to avoid name conflicts with ndarray.dtype
```

Comment:
It would be great if we could have a "strict-typing" mode that narrowed the scope of what dtype-like or array-like means. I wonder how that could be achieved...

Reply:
Yes, I'd love see this.